core/config/v2/types.go

```diff
@@ -411,6 +412,29 @@ type LogFile struct {
 	MaxBackups *int64
 }
 
+func (l *LogFile) ValidateConfig() (err error) {
```

Comment:
Note: this will catch cases where the log file has been set explicitly; it won't catch cases where the log file hasn't been set (and where we therefore default to the root dir). For those cases, the race condition will still be possible.

For that reason, I wonder if it's not better to put the EnsureDirAndMaxPerms call in the logger as I was going to originally? That is the only place where we can be sure we're operating on the right file, and can avoid the race condition.

This should also be happening sufficiently early on in the app initialization (the logger is basically the first thing that is instantiated).

Reply:
IMHO the root dir ensure call should be aligned with this and also live in a `ValidateConfig()`. Does that solve the problem?

Reply:
@jmank88 Yep, that would work! 

That means we would call EnsureDirAndMaxPerms even when instantiating a client config (albeit only if RootDir is set). Is that acceptable?

Reply:
I mean that if we put a EnsureDirAndMaxPerms call for the RootDir in a ValidateConfig() call, then we would end up running this even for "client" CLI commands (my understanding based on our slack convo was that the RootDir was a "server" setting).

eg. if we run a `./chainlink admin <...>` command, we would create a root dir.

(At the moment, the EnsureDir... call lives in the node Before() block, so wouldn't be run in this case. I just want to make sure that that's acceptable or understand if we need to work around this somehow.)

Reply:
IIRC client commands should not be invoking config validation at all though, right?

Reply:
The entire configuration interface is server side. Any remaining client side references are in error. Do we need to untangle the client logging references too though?